nbclient/util.py

```diff
@@ -89,3 +89,14 @@ async def ensure_async(obj):
         return result
     # obj doesn't need to be awaited
     return obj
+
+
+def run_hook(hook, *args):
+    if hook is None:
+        return
+    if inspect.iscoroutinefunction(hook):
+        future = hook(*args)
+    else:
+        loop = asyncio.get_event_loop()
+        future = loop.run_in_executor(None, hook, *args)
```

Comment:
Why not just call `hook(*args)` here?

Reply:
I think it'd be preferable to use kwargs so it's more forward compatible?

Reply:
I don't want to block execution, which calling hook(*args) will do, so I'm executing it on a threadpool executor

Yeah good call on the kwargs.

Reply:
OK just remembered why I went with args..
run_in_executor only takes args, not kwargs....